core/src/main/java/hudson/model/Queue.java

```diff
@@ -433,6 +441,22 @@ public synchronized void load() {
         }
     }
 
+    private long getMaxNextBuildNumber() {
+        long maxNextBuildNumber = 0;
+        Jenkins jenkins = Jenkins.getInstance();
+
+        if (jenkins != null) {
+            for (String jobName : jenkins.getJobNames()) {
+                TopLevelItem item = jenkins.getItem(jobName);
+                if (item instanceof Job) {
+                    maxNextBuildNumber = Math.max(maxNextBuildNumber, ((Job) item).getNextBuildNumber());
```

Comment:
-34343 > 123 should be true :-)

I would hope that Jenkins doesn't barf when build numbers wrap around.  Even if it does we should not be adding to that mess (people do whacky things with nextBuildNumber in plugins).